pyanaconda/modules/network/nm_client.py

```diff
@@ -1376,15 +1371,10 @@ def _get_dracut_znet_argument_from_connection(connection):
     argument = ""
     wired_setting = connection.get_setting_wired()
     if wired_setting and is_s390():
-        nettype = wired_setting.get_s390_nettype()
-        # get_s390_subchannels() returns a list of subchannels
-        subchannels = wired_setting.get_s390_subchannels()
-        if nettype and subchannels:
-            argument = "rd.znet={},{}".format(nettype, ",".join(subchannels))
-            options = wired_setting.get_property(NM.SETTING_WIRED_S390_OPTIONS)
-            if options:
-                options_string = ','.join("{}={}".format(key, val) for key, val in options.items())
-                argument += ",{}".format(options_string)
+        devspec = util.execWithCapture("/lib/s390-tools/zdev-to-rd.znet",
+                                       ["persistent",
+                                        connection.get_interface_name()]).strip()
+        argument = "rd.znet={}".format(devspec)
     return argument
```

Comment:
I think it makes no sense anymore to actually call `_get_dracut_znet_argument_from_connection()` because all it does is delegate to the external helper program called via `util.execWithCapture()`. For the sake of completeness, I added `get_interface_name` to the mock objects even though it is unused. 
Let's see if this works as I hope: https://github.com/rhinstaller/anaconda/pull/5250/commits/80c3a42087655d9700417eb1d5c7983310b3a5ae